### ISSUE TITLE
Allow retrieval of files for Custom Tasks

### ISSUE THREAD

**Issue by tnelson-doghouse** [uselagoon/lagoon-cli#234]
**Is your feature request related to a problem? Please describe.**
When I do a database dump on a non-drush site (via a Custom Task), there's no straightforward way to retrieve the backup.  

**Describe the solution you'd like**
Ben Jackson's suggestion for solving this problem (not an approval from him, just a thought) was something along the lines of:
- Make the file appear in the task view page
- Allow retrieval of the backup via `lagoon get backup --backup-id <id>`

**Describe alternatives you've considered**
Dumping the files directly into the backups list.

**Comment by rocketeerbkw** [comment]
> Make the file appear in the task view page
Not sure what this means? Task files have been shown and downloadable from the UI since we add them to the UI.

**Comment by tnelson-doghouse** [comment]
@rocketeerbkw has this changed since July of 2022?  If so, can you point me at documentation on how to take a file created by a custom task and add it to the UI?  Thanks!

**Comment by shreddedbacon** [comment]
What you originally asked is how to retrieve a file from a task, not add a file to a task

**Comment by shreddedbacon** [comment]
It is possible to upload a file to a task via the API. The CLI offers a way to [upload files to a task](https://uselagoon.github.io/lagoon-cli/commands/lagoon_upload_task-files/). 

You just need to know some information about the task. All task pods have a variable named `TASK_DATA_ID` which contains the ID that the CLI command needs.

**Comment by tnelson-doghouse** [comment]
@shreddedbacon , yes, you're right.  For context, I think this was for a database dump task; we wanted the devs to be able to dump the database, then download it from the UI.  Discussion should be visible on AIOSD-7263.  

The discussion there says that it's not currently possible to retrieve task files from the CLI.  Does that mean it'd be possible via GraphQL?  

Thanks!

**Comment by shreddedbacon** [comment]
> Does that mean it'd be possible via GraphQL?

Yes, well retrieving the download link for the file would be possible.

**Comment by tnelson-doghouse** [comment]
Oh, that makes sense.  If the CLI could do it, that'd be even better, but that's better than nothing :) .

**Comment by shreddedbacon** [comment]
> If the CLI could do it, that'd be even better

It can probably do this yes, when time permits to add the functionality I guess.

**Comment by rocketeerbkw** [comment]
> has this changed since July of 2022?

No, tasks have always been able to have files attached and downloadable from the UI.

> can you point me at documentation on how to take a file created by a custom task and add it to the UI?

We don't have documentation for this. The easiest option, as Ben suggested, is to use the lagoon cli `lagoon upload task-files --help` (this is a new option in the cli since 2022). The other option is to use a graphql query, as we do in our [native tasks using curl](https://github.com/uselagoon/lagoon/blob/main/services/api/src/resources/task/resolvers.ts#L515-L520). Both methods will upload the task to Lagoon and show a link when viewing the task.

![image](https://github.com/user-attachments/assets/ec20b64a-4dc5-4dfb-a672-0af6eb2d6cbe)

> The discussion there says that it's not currently possible to retrieve task files from the CLI. Does that mean it'd be possible via GraphQL?

This is one valid query, there are other ways of course:
```graphql
query ($taskId: Int!) {
  taskById(
    id: $taskId
  ) {
    id
    name
    files {
      id
      filename
      download
    }
  }
}
```

If you wanted to use the lagoon-cli to get the download link you could do something like the following:

`lagoon raw --raw "query {taskById(id:50476){files{download}}}" | jq -r '.taskById.files.[] | .download' | xargs curl -O`

**Comment by tnelson-doghouse** [comment]
Oh, nice!  Didn't know about lagoon raw.  Thanks!
